test/libsolidity/SemanticTest.cpp

```diff
@@ -143,12 +151,45 @@ map<string, Builtin> SemanticTest::makeBuiltins() const
 			[this](FunctionCall const& _call) -> optional<bytes>
 			{
 				soltestAssert(_call.arguments.parameters.empty(), "No arguments expected.");
-				  return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
-		 	}
+				return toBigEndian(u256(storageEmpty(m_contractAddress) ? 1 : 0));
+			}
+		},
+	};
+}
+
+vector<SideEffectHook> SemanticTest::makeSideEffectHooks() const
+{
+	return {
+		[](FunctionCall const& _call) -> vector<string>
+		{
+			if (_call.signature.find("smokeTest2") != string::npos)
+				return {_call.signature + "0"};
+			return {};
+		},
+		[](FunctionCall const& _call) -> vector<string>
+		{
+			if (_call.signature.find("smokeTest2") != string::npos)
+				return {_call.signature + "1", _call.signature + "2"};
+			return {};
 		}
 	};
 }
 
+void SemanticTest::updateSideEffects(FunctionCall const& _call)
+{
+	m_sideEffects[&_call].clear();
```

Comment:
Why do you store this in the SemanticTest class instead of inside `_call` itself?

Reply:
It is only needed locally.